pkg/combustion/kubernetes.go

```diff
@@ -22,15 +22,25 @@ const (
 	k8sDir              = "kubernetes"
 	k8sConfigDir        = "config"
 	k8sServerConfigFile = "server.yaml"
+	k8sAgentConfigFile  = "agent.yaml"
 	rke2InstallScript   = "15-rke2-install.sh"
 
+	tokenKey        = "token"
 	cniKey          = "cni"
 	cniDefaultValue = image.CNITypeCilium
+	serverKey       = "server"
+	tlsSANKey       = "tls-san"
 )
 
 var (
 	//go:embed templates/15-rke2-single-node-installer.sh.tpl
 	rke2SingleNodeInstaller string
+
+	//go:embed templates/15-rke2-multi-node-installer.sh.tpl
+	rke2MultiNodeInstaller string
+
+	//go:embed templates/rke2-vip.yaml.tpl
+	rke2VIPManifest string
```

Comment:
We may want to move this to `k8s-vip.yaml.tpl` and `k8sVIPManifest` respectively so we can adapt to k3s too; k3s will have the same requirements for the API VIP as RKE2.

Reply:
Good call out, you're entirely right. I guess we either have a specific file for rke2, or we modify the template to check if it's rke2/k3s.